File: p57.py
#solve problem 57
from math import log10

num = 1L
denom = 2L
temp = 0L
count = 0

for i in range(0,1000):
    num += 2*denom
    temp = num
    num = denom
    denom = temp

    if long(log10(num+denom))-long(log10(denom)) >= 1:
        count += 1

print count

File: p57.cpp
#include <stdio.h>
#include <iostream>
#include <math.h>

using namespace std;

void swap(unsigned long long frac[])
{
	unsigned long long temp = frac[0];
	frac[0] = frac[1];
	frac[1] = temp;
}

int solve()
{
	unsigned long long frac[] = {1,2};
	unsigned long long sqrt2[] = {1,1};
	unsigned long count = 0;
	for (int i = 0; i < 999; i++)
	{
		frac[0] += (2 * frac[1]);
		swap(frac);
//		printf("%llu/%llu\n",frac[0]+frac[1],frac[1]);
		if ((int)log10(frac[0]+frac[1])-(int)log10(frac[1]) >= 1)
			count++;
	}
	cout << count << endl;
	return count;
}

int main()
{
	solve();
	return 0;
}
